Week3/Homework/Project/Credit-Card-Validator.js

```diff
@@ -0,0 +1,90 @@
+function cardChecker(cardNumber) {
+
+    if (cardNumber * 10 % 2 !== 0) {        // Check if cardNumber is not a number (if it has letters or symbols)
+
+        const newCardNumber = prompt("Invalid Characters, Please Enter Just Numbers :");
+        return cardChecker(newCardNumber);
+
+    }
+
+    else if (cardNumber * 10 % 2 === 0) {       // Check if cardnumber is a number
+
+        const copyOfCardNumber = cardNumber;
+
+        if (copyOfCardNumber.length !== 16) {
+
+            const newCardNumber = prompt("The card number that you entered has not 16 digits, Please Enter a Valid Card Number :");
+            return cardChecker(newCardNumber);
+
+        }
+
+        else if (copyOfCardNumber.length === 16) {      // Check if copyOfCardNumber has 16 digits
+
+            for (let i = 1; i < copyOfCardNumber.length; i++) {     // This loop compare the first digit of copyOfCardNumber with the all other 15 digits 
+
+                if (copyOfCardNumber[0] === copyOfCardNumber[i]) {
+
+                    if (i === copyOfCardNumber.length - 1) {    // Check if it reached to the last digit of copyOfCardNumber
+
+                        const newCardNumber = prompt("The card number that you entered has the same 16 digits, Please Enter a Valid Card Number :");
+                        return cardChecker(newCardNumber);
+
+                    }
+
+                }
+
+                else {
+
+                    break;
+
+                }
+
+            }
+
+            if (copyOfCardNumber[copyOfCardNumber.length - 1] % 2 !== 0) {      // Check if the last digit of copyOfCardNumber is ood
+
+                const newCardNumber = prompt("The last digit of your card number is not even, Please Enter a Valid Card Number :");
+                return cardChecker(newCardNumber);
+
+            }
+
+            else {                                                               // if the last digit of copyOfCardNumber is evev
+
+                let sumOfdigits = 0;
+
+                for (let i = 0; i < copyOfCardNumber.length; i++) {
+
+                    const digitValue = copyOfCardNumber[i] * 1;
+                    sumOfdigits = sumOfdigits + digitValue;
+
+                }
```

Comment:
Try using `reduce` here instead of summing using a for loop. 🙂 

Also, I see a lot of empty lines in your code. I presume you added these to make your code more readable but it actually serves the opposite purpose. We don't really need new lines separating pieces of our code as it makes it harder to read 🙂